rflx/pyrflx/typevalue.py

```diff
@@ -887,17 +1019,40 @@ def __update_simplified_mapping(self) -> None:
             **{First(k): v.first for k, v in self._fields.items() if v.set},
             **{Last(k): v.last for k, v in self._fields.items() if v.set},
         }
+
         self._simplified_mapping = {**field_values, **self.__type_literals}
 
+        pre_final = self._prev_field("Final")
+        if pre_final != "" and self._fields[pre_final].set:
+            self._simplified_mapping[Last("Message")] = self._fields[pre_final].last
+
     def __simplified(self, expr: Expr) -> Expr:
         if not self._simplified_mapping:
             self.__update_simplified_mapping()
+
         return (
             expr.substituted(mapping=self._simplified_mapping)
             .substituted(mapping=self._simplified_mapping)
             .simplified()
         )
 
+    class Checksum:
+        def __init__(self, checksum_field_name: str, expressions: Sequence[Expr]):
+            self.field_name: str = checksum_field_name
+            self.function: Optional[Callable] = None
+            self.parameters: List["MessageValue.Checksum.ExpressionTuple"] = []
+            for expr in expressions:
+                assert isinstance(expr, (ValueRange, Attribute, Variable))
+                self.parameters.append(self.ExpressionTuple(expr))
+
+        def set_checksum_function(self, function: Callable) -> None:
+            self.function = function
+
+        @dataclass
+        class ExpressionTuple:
+            expression: Expr
+            evaluated_expression: Expr = UNDEFINED
```

Comment:
Is there any reason to declare this at the end of the class?

Reply:
Moved to top of the class in 3b0916c